Classes/Hooks/HookService.php

```diff
@@ -0,0 +1,129 @@
+<?php
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Hooks;
+
+use class_implements;
+use defined;
+use get_class;
+use in_array;
+use interface_exists;
+
+use UnexpectedValueException;
+
+use OliverKlee\Seminars\Interfaces\Hook;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * This class provides functions for unified hooks.
+ *
+ * It refactors the wide-spread use of this type of hooks in seminars.
+ *
+ * @author Oliver Klee <typo3-coding@oliverklee.de>
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+class HookService
```

Comment:
The job of this class is to find/retrieve the hooks. What do you think about changing the name to `HookFinder`?

Reply:
In my eyes a finder uses some kind of wildcard to auto-magically find something. The Service was intended to calling the hooks in addition to only provide a collection of them. I haven't implemented that yet due to a lack of time.

Reply:
HookProvider?

Reply:
> HookProvider?

Sounds good to me.